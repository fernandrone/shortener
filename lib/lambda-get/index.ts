import { DynamoDB } from 'aws-sdk';

const ddb = new DynamoDB.DocumentClient();
const fallback = 'https://fernandrone.com';

export const handler = async (event: any = {}): Promise<any> => {
  console.log(event);

  if (!('path' in event) || event.path.length < 2) {
    return redirect(fallback);
  }

  const params = {
    TableName: process.env.DYNAMODB_TABLE!,
    Key: event.path,
  };

  ddb.get(params, function(err, data) {
    if (err) {
      console.log(err);
      return redirect(fallback);
    }

    if (data.Item === undefined) {
      return redirect(fallback);
    }

    return redirect(data.Item['id']);
  });
};

function redirect(url: string) {
  return {
    statusCode: 302,
    headers: {
      Location: url,
    },
  };
}
