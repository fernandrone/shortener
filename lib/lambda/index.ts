export const handler = async (event: any = {}): Promise<any> => {
  const response = {
    statusCode: 302,
    headers: {
      Location: "https://fernandrone.com"
    }
  };
  return response;
};
