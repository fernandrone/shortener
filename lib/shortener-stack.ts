import * as apigateway from '@aws-cdk/aws-apigateway';
import * as dynamodb from '@aws-cdk/aws-dynamodb';
import * as core from '@aws-cdk/core';
import * as certmgr from '@aws-cdk/aws-certificatemanager';
import * as lambda from '@aws-cdk/aws-lambda';
import * as path from 'path';
import * as route53 from '@aws-cdk/aws-route53';
import * as targets from '@aws-cdk/aws-route53-targets';

export class ShortenerStack extends core.Stack {
  readonly domain = 'fdr.one';
  readonly stage = 'prod';
  readonly origin = 'https://' + this.domain;

  constructor(scope: core.App, id: string, props?: core.StackProps) {
    super(scope, id, props);

    const api = new apigateway.RestApi(this, 'shortenerAPI', {
      // these will limit burst of requests that might increase our billing too much
      deployOptions: {
        stageName: this.stage,
        throttlingRateLimit: 5,
        throttlingBurstLimit: 10,
      },
      defaultCorsPreflightOptions: {
        allowOrigins: [this.origin],
      },
    });

    const hostedZone = route53.HostedZone.fromLookup(this, 'shortenerHostedZone', {
      domainName: this.domain,
      privateZone: false,
    });

    const certificate = new certmgr.DnsValidatedCertificate(this, 'shortenerCert', {
      domainName: this.domain,
      hostedZone,
    });

    const customDomain = new apigateway.DomainName(this, 'shortenerCustomDomain', {
      domainName: this.domain,
      certificate: certificate,
      endpointType: apigateway.EndpointType.EDGE,
    });

    customDomain.addBasePathMapping(api);

    new route53.ARecord(this, 'shortenerAliasRecord', {
      zone: hostedZone,
      target: route53.RecordTarget.fromAlias(
        new targets.ApiGatewayDomain(customDomain),
      ),
    });

    const table = new dynamodb.Table(this, 'shortenerTable', {
      partitionKey: { name: 'id', type: dynamodb.AttributeType.STRING },
      sortKey: { name: 'url', type: dynamodb.AttributeType.STRING },
      billingMode: dynamodb.BillingMode.PROVISIONED,
      readCapacity: 5,
      writeCapacity: 1,
    });

    const getFnc = new lambda.Function(this, 'shortenerBackend', {
      runtime: lambda.Runtime.NODEJS_12_X,
      handler: 'index.handler',
      code: lambda.Code.fromAsset(path.join(__dirname, 'lambda-get')),
      environment: {
        DYNAMODB_TABLE: table.tableName,
      },
    });

    // assign the 'lambda-get' function to the GET method
    const integration = new apigateway.LambdaIntegration(getFnc, {
      proxy: true,
    });

    // add greedy proxy
    const proxy = api.root.addProxy({
      defaultIntegration: integration,
      anyMethod: false,
    });

    proxy.addMethod('GET', integration);

    // allow the lambda function to r/w date into the dynamodb table
    table.grantReadData(getFnc);

    new core.CfnOutput(this, 'shortenerOrigin', {
      value: this.origin,
      description: 'The value of the custom shortener origin',
      exportName: 'ShortenerCustomOrigin',
    });

    core.Tag.add(this, 'project', 'shortener');
  }
}
