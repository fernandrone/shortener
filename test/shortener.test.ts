import { ShortenerStack } from '../lib/shortener-stack';
import { SynthUtils } from '@aws-cdk/assert';
import * as cdk from '@aws-cdk/core';
import '@aws-cdk/assert/jest';

const params = {
  env: {
    account: 'dummy',
    region: 'us-west-2',
  },
};

test('Lambda Function Created', () => {
  const app = new cdk.App();
  // WHEN
  const stack = new ShortenerStack(app, 'MyTestStack', params);
  // THEN
  expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
});

test('Lambda Function matches Runtime', () => {
  const app = new cdk.App();
  // WHEN
  const stack = new ShortenerStack(app, 'MyTestStack', params);
  // THEN
  expect(stack).toHaveResource('AWS::Lambda::Function', {
    Runtime: 'nodejs12.x',
  });
});
