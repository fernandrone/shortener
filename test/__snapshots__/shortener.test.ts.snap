// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lambda Function Created 1`] = `
Object {
  "Outputs": Object {
    "shortenerAPIEndpointD2575410": Object {
      "Value": Object {
        "Fn::Join": Array [
          "",
          Array [
            "https://",
            Object {
              "Ref": "shortenerAPID8E48A72",
            },
            ".execute-api.us-west-2.",
            Object {
              "Ref": "AWS::URLSuffix",
            },
            "/",
            Object {
              "Ref": "shortenerAPIDeploymentStageprod26C06356",
            },
            "/",
          ],
        ],
      },
    },
    "shortenerOrigin": Object {
      "Description": "The value of the custom shortener origin",
      "Export": Object {
        "Name": "ShortenerCustomOrigin",
      },
      "Value": "https://fdr.one",
    },
  },
  "Parameters": Object {
    "AssetParameters0b9723d54b7db3fbfc1a143398b02392e5fe080a68535480782a949b4372d0f0ArtifactHashF8F836D1": Object {
      "Description": "Artifact hash for asset \\"0b9723d54b7db3fbfc1a143398b02392e5fe080a68535480782a949b4372d0f0\\"",
      "Type": "String",
    },
    "AssetParameters0b9723d54b7db3fbfc1a143398b02392e5fe080a68535480782a949b4372d0f0S3Bucket0997A4A0": Object {
      "Description": "S3 bucket for asset \\"0b9723d54b7db3fbfc1a143398b02392e5fe080a68535480782a949b4372d0f0\\"",
      "Type": "String",
    },
    "AssetParameters0b9723d54b7db3fbfc1a143398b02392e5fe080a68535480782a949b4372d0f0S3VersionKey83D9C166": Object {
      "Description": "S3 key for asset version \\"0b9723d54b7db3fbfc1a143398b02392e5fe080a68535480782a949b4372d0f0\\"",
      "Type": "String",
    },
    "AssetParametersc2945fd6904fced6a00c825b02118353df436c7378af32454b476d0ec8025705ArtifactHash6E20F2F1": Object {
      "Description": "Artifact hash for asset \\"c2945fd6904fced6a00c825b02118353df436c7378af32454b476d0ec8025705\\"",
      "Type": "String",
    },
    "AssetParametersc2945fd6904fced6a00c825b02118353df436c7378af32454b476d0ec8025705S3Bucket70659BF0": Object {
      "Description": "S3 bucket for asset \\"c2945fd6904fced6a00c825b02118353df436c7378af32454b476d0ec8025705\\"",
      "Type": "String",
    },
    "AssetParametersc2945fd6904fced6a00c825b02118353df436c7378af32454b476d0ec8025705S3VersionKey5A60729D": Object {
      "Description": "S3 key for asset version \\"c2945fd6904fced6a00c825b02118353df436c7378af32454b476d0ec8025705\\"",
      "Type": "String",
    },
  },
  "Resources": Object {
    "shortenerAPIAccount4DD89EBF": Object {
      "DependsOn": Array [
        "shortenerAPID8E48A72",
      ],
      "Properties": Object {
        "CloudWatchRoleArn": Object {
          "Fn::GetAtt": Array [
            "shortenerAPICloudWatchRole8A79B60B",
            "Arn",
          ],
        },
      },
      "Type": "AWS::ApiGateway::Account",
    },
    "shortenerAPICloudWatchRole8A79B60B": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "apigateway.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AmazonAPIGatewayPushToCloudWatchLogs",
              ],
            ],
          },
        ],
        "Tags": Array [
          Object {
            "Key": "project",
            "Value": "shortener",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "shortenerAPID8E48A72": Object {
      "Properties": Object {
        "Name": "shortenerAPI",
        "Tags": Array [
          Object {
            "Key": "project",
            "Value": "shortener",
          },
        ],
      },
      "Type": "AWS::ApiGateway::RestApi",
    },
    "shortenerAPIDeployment1BF02EFDf0d5ebe8f606dbaa1721373ccbb54599": Object {
      "DependsOn": Array [
        "shortenerAPIGET9DBD2395",
        "shortenerAPIOPTIONS8F59B854",
      ],
      "Properties": Object {
        "Description": "Automatically created by the RestApi construct",
        "RestApiId": Object {
          "Ref": "shortenerAPID8E48A72",
        },
      },
      "Type": "AWS::ApiGateway::Deployment",
    },
    "shortenerAPIDeploymentStageprod26C06356": Object {
      "Properties": Object {
        "DeploymentId": Object {
          "Ref": "shortenerAPIDeployment1BF02EFDf0d5ebe8f606dbaa1721373ccbb54599",
        },
        "MethodSettings": Array [
          Object {
            "HttpMethod": "*",
            "ResourcePath": "/*",
            "ThrottlingBurstLimit": 10,
            "ThrottlingRateLimit": 5,
          },
        ],
        "RestApiId": Object {
          "Ref": "shortenerAPID8E48A72",
        },
        "StageName": "prod",
        "Tags": Array [
          Object {
            "Key": "project",
            "Value": "shortener",
          },
        ],
      },
      "Type": "AWS::ApiGateway::Stage",
    },
    "shortenerAPIGET9DBD2395": Object {
      "Properties": Object {
        "AuthorizationType": "NONE",
        "HttpMethod": "GET",
        "Integration": Object {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:us-west-2:lambda:path/2015-03-31/functions/",
                Object {
                  "Fn::GetAtt": Array [
                    "shortenerBackendBFAB597E",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": Object {
          "Fn::GetAtt": Array [
            "shortenerAPID8E48A72",
            "RootResourceId",
          ],
        },
        "RestApiId": Object {
          "Ref": "shortenerAPID8E48A72",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "shortenerAPIGETApiPermissionMyTestStackshortenerAPIA32CAD80GETB5E72F51": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "shortenerBackendBFAB597E",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:us-west-2:dummy:",
              Object {
                "Ref": "shortenerAPID8E48A72",
              },
              "/",
              Object {
                "Ref": "shortenerAPIDeploymentStageprod26C06356",
              },
              "/GET/",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "shortenerAPIGETApiPermissionTestMyTestStackshortenerAPIA32CAD80GET47CCA933": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "shortenerBackendBFAB597E",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:us-west-2:dummy:",
              Object {
                "Ref": "shortenerAPID8E48A72",
              },
              "/test-invoke-stage/GET/",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "shortenerAPIOPTIONS8F59B854": Object {
      "Properties": Object {
        "AuthorizationType": "NONE",
        "HttpMethod": "OPTIONS",
        "Integration": Object {
          "IntegrationResponses": Array [
            Object {
              "ResponseParameters": Object {
                "method.response.header.Access-Control-Allow-Headers": "'Content-Type,X-Amz-Date,Authorization,X-Api-Key,X-Amz-Security-Token,X-Amz-User-Agent'",
                "method.response.header.Access-Control-Allow-Methods": "'OPTIONS,GET,PUT,POST,DELETE,PATCH,HEAD'",
                "method.response.header.Access-Control-Allow-Origin": "'https://fdr.one'",
                "method.response.header.Vary": "'Origin'",
              },
              "StatusCode": "204",
            },
          ],
          "RequestTemplates": Object {
            "application/json": "{ statusCode: 200 }",
          },
          "Type": "MOCK",
        },
        "MethodResponses": Array [
          Object {
            "ResponseParameters": Object {
              "method.response.header.Access-Control-Allow-Headers": true,
              "method.response.header.Access-Control-Allow-Methods": true,
              "method.response.header.Access-Control-Allow-Origin": true,
              "method.response.header.Vary": true,
            },
            "StatusCode": "204",
          },
        ],
        "ResourceId": Object {
          "Fn::GetAtt": Array [
            "shortenerAPID8E48A72",
            "RootResourceId",
          ],
        },
        "RestApiId": Object {
          "Ref": "shortenerAPID8E48A72",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "shortenerAliasRecordDDE56FDF": Object {
      "Properties": Object {
        "AliasTarget": Object {
          "DNSName": Object {
            "Fn::GetAtt": Array [
              "shortenerCustomDomainDE8FD5C8",
              "DistributionDomainName",
            ],
          },
          "HostedZoneId": Object {
            "Fn::GetAtt": Array [
              "shortenerCustomDomainDE8FD5C8",
              "DistributionHostedZoneId",
            ],
          },
        },
        "HostedZoneId": "DUMMY",
        "Name": "fdr.one.",
        "Type": "A",
      },
      "Type": "AWS::Route53::RecordSet",
    },
    "shortenerBackendBFAB597E": Object {
      "DependsOn": Array [
        "shortenerBackendServiceRoleDefaultPolicy7DD91C30",
        "shortenerBackendServiceRoleD10FFAB4",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParametersc2945fd6904fced6a00c825b02118353df436c7378af32454b476d0ec8025705S3Bucket70659BF0",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParametersc2945fd6904fced6a00c825b02118353df436c7378af32454b476d0ec8025705S3VersionKey5A60729D",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParametersc2945fd6904fced6a00c825b02118353df436c7378af32454b476d0ec8025705S3VersionKey5A60729D",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Environment": Object {
          "Variables": Object {
            "DYNAMODB_TABLE": Object {
              "Ref": "shortenerTable025392D9",
            },
          },
        },
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "shortenerBackendServiceRoleD10FFAB4",
            "Arn",
          ],
        },
        "Runtime": "nodejs12.x",
        "Tags": Array [
          Object {
            "Key": "project",
            "Value": "shortener",
          },
        ],
      },
      "Type": "AWS::Lambda::Function",
    },
    "shortenerBackendServiceRoleD10FFAB4": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
        "Tags": Array [
          Object {
            "Key": "project",
            "Value": "shortener",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "shortenerBackendServiceRoleDefaultPolicy7DD91C30": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "dynamodb:BatchGetItem",
                "dynamodb:GetRecords",
                "dynamodb:GetShardIterator",
                "dynamodb:Query",
                "dynamodb:GetItem",
                "dynamodb:Scan",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "shortenerTable025392D9",
                    "Arn",
                  ],
                },
                Object {
                  "Ref": "AWS::NoValue",
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "shortenerBackendServiceRoleDefaultPolicy7DD91C30",
        "Roles": Array [
          Object {
            "Ref": "shortenerBackendServiceRoleD10FFAB4",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "shortenerCertCertificateRequestorFunctionE626283C": Object {
      "DependsOn": Array [
        "shortenerCertCertificateRequestorFunctionServiceRoleDefaultPolicy8E443F6F",
        "shortenerCertCertificateRequestorFunctionServiceRole880ADB54",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParameters0b9723d54b7db3fbfc1a143398b02392e5fe080a68535480782a949b4372d0f0S3Bucket0997A4A0",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters0b9723d54b7db3fbfc1a143398b02392e5fe080a68535480782a949b4372d0f0S3VersionKey83D9C166",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters0b9723d54b7db3fbfc1a143398b02392e5fe080a68535480782a949b4372d0f0S3VersionKey83D9C166",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Handler": "index.certificateRequestHandler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "shortenerCertCertificateRequestorFunctionServiceRole880ADB54",
            "Arn",
          ],
        },
        "Runtime": "nodejs10.x",
        "Tags": Array [
          Object {
            "Key": "project",
            "Value": "shortener",
          },
        ],
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "shortenerCertCertificateRequestorFunctionServiceRole880ADB54": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
        "Tags": Array [
          Object {
            "Key": "project",
            "Value": "shortener",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "shortenerCertCertificateRequestorFunctionServiceRoleDefaultPolicy8E443F6F": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "acm:RequestCertificate",
                "acm:DescribeCertificate",
                "acm:DeleteCertificate",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
            Object {
              "Action": "route53:GetChange",
              "Effect": "Allow",
              "Resource": "*",
            },
            Object {
              "Action": "route53:changeResourceRecordSets",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:",
                    Object {
                      "Ref": "AWS::Partition",
                    },
                    ":route53:::hostedzone/DUMMY",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "shortenerCertCertificateRequestorFunctionServiceRoleDefaultPolicy8E443F6F",
        "Roles": Array [
          Object {
            "Ref": "shortenerCertCertificateRequestorFunctionServiceRole880ADB54",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "shortenerCertCertificateRequestorResource9C94F867": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "DomainName": "fdr.one",
        "HostedZoneId": "DUMMY",
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "shortenerCertCertificateRequestorFunctionE626283C",
            "Arn",
          ],
        },
      },
      "Type": "AWS::CloudFormation::CustomResource",
      "UpdateReplacePolicy": "Delete",
    },
    "shortenerCustomDomainDE8FD5C8": Object {
      "Properties": Object {
        "CertificateArn": Object {
          "Fn::GetAtt": Array [
            "shortenerCertCertificateRequestorResource9C94F867",
            "Arn",
          ],
        },
        "DomainName": "fdr.one",
        "EndpointConfiguration": Object {
          "Types": Array [
            "EDGE",
          ],
        },
        "Tags": Array [
          Object {
            "Key": "project",
            "Value": "shortener",
          },
        ],
      },
      "Type": "AWS::ApiGateway::DomainName",
    },
    "shortenerCustomDomainMapMyTestStackshortenerAPIA32CAD8091043B39": Object {
      "Properties": Object {
        "DomainName": Object {
          "Ref": "shortenerCustomDomainDE8FD5C8",
        },
        "RestApiId": Object {
          "Ref": "shortenerAPID8E48A72",
        },
        "Stage": Object {
          "Ref": "shortenerAPIDeploymentStageprod26C06356",
        },
      },
      "Type": "AWS::ApiGateway::BasePathMapping",
    },
    "shortenerTable025392D9": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "AttributeDefinitions": Array [
          Object {
            "AttributeName": "id",
            "AttributeType": "S",
          },
        ],
        "KeySchema": Array [
          Object {
            "AttributeName": "id",
            "KeyType": "HASH",
          },
        ],
        "ProvisionedThroughput": Object {
          "ReadCapacityUnits": 5,
          "WriteCapacityUnits": 1,
        },
        "TableName": "shortenedURLs",
        "Tags": Array [
          Object {
            "Key": "project",
            "Value": "shortener",
          },
        ],
      },
      "Type": "AWS::DynamoDB::Table",
      "UpdateReplacePolicy": "Retain",
    },
  },
}
`;
